src/transaction/signTransaction.js

```diff
@@ -20,15 +20,43 @@ import serializeTransactionIntoCanonicalString from './serializeTransactionIntoC
  */
 export default function signTransaction(transaction, ...privateKeys) {
     const signedTx = clone(transaction)
-    signedTx.inputs.forEach((input, index) => {
-        const privateKey = privateKeys[index]
-        const privateKeyBuffer = new Buffer(base58.decode(privateKey))
-        const serializedTransaction = serializeTransactionIntoCanonicalString(transaction)
-        const ed25519Fulfillment = new cc.Ed25519Sha256()
-        ed25519Fulfillment.sign(new Buffer(serializedTransaction), privateKeyBuffer)
-        const fulfillmentUri = ed25519Fulfillment.serializeUri()
-
-        input.fulfillment = fulfillmentUri
+    transaction.inputs.forEach((input) => {
+        input.fulfillment = null // OJOOO
+    })
+    const serializedTransaction = serializeTransactionIntoCanonicalString(transaction)
+    signedTx.inputs.forEach((input) => {
+        if (input.fulfillment.type === 'ed25519-sha-256') {
+            const privateKey = privateKeys[0]
+            privateKeys.splice(0, 1)
+            const privateKeyBuffer = new Buffer(base58.decode(privateKey))
+            const ed25519Fulfillment = new cc.Ed25519Sha256()
+            ed25519Fulfillment.sign(new Buffer(serializedTransaction), privateKeyBuffer)
+            const fulfillmentUri = ed25519Fulfillment.serializeUri()
+            input.fulfillment = fulfillmentUri
+        } else if (input.fulfillment.type === 'threshold-sha-256') {
+            // Not valid for more than one input with m-of-n signatures where m < n.
+            const thresholdFulfillment = new cc.ThresholdSha256()
+            // m represents the number of signatures needed for this input
+            let m = 0
+            input.fulfillment.subconditions.forEach((subcdt) => {
```

Comment:
`subcdts` is not an appreviation we commonly use in the code base.
I suggest to rename to a word we also use in other implementations.